nmostesting/suites/IS1101Test.py

```diff
@@ -0,0 +1,1725 @@
+        if response.status_code != 200:
+            return test.FAIL(
+                "The request has not succeeded: {}".format(response.json())
+            )
+        self.senders = response.json()
+        if len(self.senders) == 0:
+            return test.UNCLEAR("There is no IS-11 senders.")
+        return test.PASS()
+
+    def test_02_01_01(self, test):
+        "Verify that the device supports the concept of IS-11 Sender"
+        if len(self.senders) != 0:
+            for sender_id in self.senders:
+                _, response = TestHelper.do_request(
+                    "GET", self.node_url + "senders/" + sender_id
+                )
+                if response.status_code != 200:
+                    return test.FAIL(
+                        "The request has not succeeded: {}".format(response.json())
+                    )
+                sender_node = response.json()["id"]
+                if sender_id[:-1] != sender_node:
+                    return test.FAIL("Senders are different")
+            return test.PASS()
+        return test.UNCLEAR("There is no IS-11 senders.")
+
+    def test_02_02(self, test):
+        "Verify senders (generic with/without inputs)"
+        _, response = TestHelper.do_request("GET", self.compat_url + "senders/")
+        if response.status_code != 200:
+            return test.FAIL(
+                "The request has not succeeded: {}".format(response.json())
+            )
+        self.senders_2 = response.json()
+        if len(self.senders_2) == 0:
+            return test.UNCLEAR("There is no IS-11 senders.")
+        return test.PASS()
+
+    def test_02_02_01(self, test):
+        "Verify that the status is unconstrained as per our pre-conditions"
+        if len(self.senders_2) != 0:
+            for sender_id in self.senders_2:
+                _, response = TestHelper.do_request(
+                    "GET", self.compat_url + "senders/" + sender_id + "status/"
+                )
+                if response.status_code != 200:
+                    return test.FAIL(
+                        "The request has not succeeded: {}".format(response.json())
+                    )
+                state = response.json()["state"]
+                if state in ["awating_essence", "no_essence"]:
+                    for i in range(0, 5):
+                        _, response = TestHelper.do_request(
+                            "GET", self.compat_url + "senders/" + sender_id + "status/"
+                        )
+                        if response.status_code != 200:
+                            return test.FAIL(
+                                "The request has not succeeded: {}".format(
+                                    response.json()
+                                )
+                            )
+                        state = response.json()["state"]
+                        if state in ["awating_essence", "no_essence"]:
+                            time.sleep(3)
```

Comment:
Is maximum 5 x 3 seconds sleep part of the spec? If not,  should probably be configurable?

Reply:
This not not part of the IS-11 specification ... just an expectation that the device somehow is responsive within a few seconds.